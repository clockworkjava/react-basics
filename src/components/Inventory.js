import React from 'react';
import BookView from './BookView';
import {fbase} from '../fbase';

class Inventory extends React.Component {

    constructor() {
        super();
        this.state = {
            books: []
        }
    }

    componentDidMount() {
        this.ref = fbase.syncState('bookstore/books',{
            context: this,
            state: 'books'
        });
   }

   componentWillUnmount() {
       fbase.removeBinding(this.ref);
   }

    render() {
        const bookListing = this.state.books.map( book => {
           return <BookView book={book} addToOrder={this.props.addToOrder}/>
        });

        return (
            <div className="inventory col-xs-6">
                <h2>Bookstore inventory:</h2>
                {bookListing}
            </div>
        );
    }
}

export default Inventory;