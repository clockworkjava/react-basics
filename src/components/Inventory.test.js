import React from 'react';
import ReactDOM from 'react-dom';
import Inventory from './Inventory';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Inventory tests', () => {

    it('Inventory renders without a problem', () => {
        const div = document.createElement('div');
        const books = [];
        ReactDOM.render(<Inventory books={books}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('Snapshot matches', () => {
        const books = [];
        const wrapper = shallow(<Inventory books={books}/>);
        expect(wrapper).toMatchSnapshot();
    })

})

