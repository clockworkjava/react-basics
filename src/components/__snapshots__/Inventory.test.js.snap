// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory tests Snapshot matches 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Inventory
    books={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Bookstore inventory:
</h2>,
        Array [],
      ],
      "className": "inventory col-xs-6",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bookstore inventory:",
        },
        "ref": null,
        "rendered": "Bookstore inventory:",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Bookstore inventory:
</h2>,
          Array [],
        ],
        "className": "inventory col-xs-6",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bookstore inventory:",
          },
          "ref": null,
          "rendered": "Bookstore inventory:",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
